### PR TITLE
perf: decode proposal id from tx resp

### PR DESCRIPTION
**Pull request by duvbell**
## Description

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/testsuite/tx.go [hunk 1]
```diff
@@ -222,16 +222,13 @@ func (s *E2ETestSuite) ExecuteAndPassGovV1Beta1Proposal(ctx context.Context, cha
 		panic("ExecuteAndPassGovV1Beta1Proposal must be passed a cosmos.CosmosChain")
 	}
 
-	proposalID := s.proposalIDs[chain.Config().ChainID]
-	defer func() {
```
Comment (gjermundgaraba) [inline e2e/testsuite/tx.go:226]: I still think this is used somewhere else, so unless we can get rid of the whole s.proposalIDs, I think we still need the defer below line 231.

If you can easily get the proposalID for the other use and get rid of it entirely, that is preferable :)

**Review by gjermundgaraba — CHANGES_REQUESTED** [review]
Thanks for the PR! This is much better, for sure :)

I left a small comment on the use of the suite's `proposalIDs` which I think we need to resolve before we can merge.

File: e2e/testsuite/tx.go (hunk 1 above)
Comment (duvbell) [reply e2e/testsuite/tx.go:226]: Hello @gjermundgaraba , i updated, can you help me review it

Comment (gjermundgaraba) [reply e2e/testsuite/tx.go:226]: This looks correct to me! Thank you :)

**Review by gjermundgaraba — APPROVED** [review]

**Pull request merged**